pkg/controller/database/dbsubnetgroup/controller.go

```diff
@@ -18,175 +18,215 @@ package dbsubnetgroup
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.Status.SetConditions(runtimev1alpha1.Creating())
-
+	cr.SetConditions(runtimev1alpha1.Creating())
 	input := &awsrds.CreateDBSubnetGroupInput{
-		DBSubnetGroupDescription: aws.String(cr.Spec.DBSubnetGroupDescription),
-		DBSubnetGroupName:        aws.String(cr.Spec.DBSubnetGroupName),
-		SubnetIds:                cr.Spec.SubnetIDs,
-		Tags:                     []awsrds.Tag{},
+		DBSubnetGroupDescription: aws.String(cr.Spec.ForProvider.Description),
+		DBSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		SubnetIds:                cr.Spec.ForProvider.SubnetIDs,
 	}
 
-	for _, t := range cr.Spec.Tags {
-		input.Tags = append(input.Tags, awsrds.Tag{
-			Key:   aws.String(t.Key),
-			Value: aws.String(t.Value),
-		})
+	if len(cr.Spec.ForProvider.Tags) != 0 {
+		input.Tags = make([]awsrds.Tag, len(cr.Spec.ForProvider.Tags))
+		for i, val := range cr.Spec.ForProvider.Tags {
+			input.Tags[i] = awsrds.Tag{Key: aws.String(val.Key), Value: aws.String(val.Value)}
+		}
 	}
 
-	req := e.client.CreateDBSubnetGroupRequest(input)
-
-	response, err := req.Send(ctx)
+	_, err := e.client.CreateDBSubnetGroupRequest(input).Send(ctx)
+	return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+}
 
-	if err != nil {
-		return managed.ExternalCreation{}, errors.Wrapf(err, errCreate, cr.Spec.DBSubnetGroupName)
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1beta1.DBSubnetGroup)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
 	}
 
-	cr.UpdateExternalStatus(*response.DBSubnetGroup)
+	_, err := e.client.ModifyDBSubnetGroupRequest(&awsrds.ModifyDBSubnetGroupInput{
+		DBSubnetGroupName:        aws.String(meta.GetExternalName(cr)),
+		DBSubnetGroupDescription: aws.String(cr.Spec.ForProvider.Description),
+		SubnetIds:                cr.Spec.ForProvider.SubnetIDs,
+	}).Send(ctx)
 
-	return managed.ExternalCreation{}, nil
-}
-
-func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
-	// TODO(soorena776): add more sophisticated Update logic, once we
-	// categorize immutable vs mutable fields (see #727)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
 
-	return managed.ExternalUpdate{}, nil
+	if len(cr.Spec.ForProvider.Tags) > 0 {
+		tags := make([]awsrds.Tag, len(cr.Spec.ForProvider.Tags))
+		for i, t := range cr.Spec.ForProvider.Tags {
+			tags[i] = awsrds.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
+		}
+		_, err = e.client.AddTagsToResourceRequest(&awsrds.AddTagsToResourceInput{
+			ResourceName: aws.String(cr.Status.AtProvider.ARN),
+			Tags:         tags,
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errAddTagsFailed)
+		}
+	}
+	return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
```

Comment:
Nitpick: `err` is from `ModifyDBSubnetGroupRequest`, which is irrelevant at this point. But it's same functionally, since that `err` is `nil` at that point anyway.